pkg/sdn/api/types.go

```diff
@@ -60,3 +60,33 @@ type NetNamespaceList struct {
 	unversioned.ListMeta
 	Items []NetNamespace
 }
+
+// EgressFirewallPolicy gives the type of an EgressFirewallRule
+type EgressFirewallPolicy string
+
+const (
+	EgressFirewallPolicyAllow EgressFirewallPolicy = "allow"
+	EgressFirewallPolicyDeny  EgressFirewallPolicy = "deny"
+)
+
+// EgressFirewallRule contains a single outgoing traffic firewall rule
+type EgressFirewallRule struct {
+	Policy EgressFirewallPolicy
+	From   string // Namespace name or empty
```

Comment:
On Jul 15, 2016, at 10:14 AM, Dan Winship <notifications@github.com> wrote:

In pkg/sdn/api/types.go
<https://github.com/openshift/origin/pull/9227#discussion_r70979582>:

> @@ -60,3 +60,33 @@ type NetNamespaceList struct {
>  	unversioned.ListMeta
>  	Items []NetNamespace
>  }
> +
> +// EgressFirewallPolicy gives the type of an EgressFirewallRule
> +type EgressFirewallPolicy string
> +
> +const (
> +	EgressFirewallPolicyAllow EgressFirewallPolicy = "allow"
> +	EgressFirewallPolicyDeny  EgressFirewallPolicy = "deny"
> +)
> +
> +// EgressFirewallRule contains a single outgoing traffic firewall rule
> +type EgressFirewallRule struct {
> +	Policy EgressFirewallPolicy
> +	From   string // Namespace name or empty

To be consistent with the upstream proposals so far, we would make
EgressPolicyPeer be the To field, not the From field.

And to be consistent with the upstream proposals, From would not exist at
all: instead, we would allow creating
EgressFirewall/EgressNetworkPolicy/whatever
objects in any namespace, and each object's rules would apply to traffic
coming from the namespace that that object is in. (They would have to still
be only creatable by cluster admins though, not project admins.)



That's generally what we do anyway - quota and limit apply to their
namespace, and they are only editable by cluster admin.


We would still need somewhere to specify the final fallback/override rules
for all remaining traffic (from all namespaces) that wasn't explicitly
accepted or denied by any previous rule. None of the upstream proposals so
far have attempted to handle that though, so there's no precedent to
follow. We could use the default namespace for that, which kind of makes
sense (and kind of doesn't). ?


Do we need fallback rules?  For quota we simply distribute the rules to all
namespaces.  For other places we create cluster scoped resources or use
global config.

If we think in the short term we don't need project admins editing egress
rules, then let's just have no fallback, and require admins to specify them
(we already effectively do this for quota, so again, no different).


—
You are receiving this because you are on a team that was mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/9227/files/7d5bf0193c381259ef2b0bc4170d5cc76a113231#r70979582>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/ABG_pzu2EBxB6RFg8H4bJHcDs8tXMdb7ks5qV5XWgaJpZM4IxESt>
.